backend/src/main/java/com/htecgroup/skynest/model/entity/UserObjectAccessKey.java

```diff
@@ -4,22 +4,22 @@
 import lombok.EqualsAndHashCode;
 import lombok.NoArgsConstructor;
 
-import javax.persistence.JoinColumn;
-import javax.persistence.ManyToOne;
+import javax.persistence.Column;
+import javax.persistence.Embeddable;
 import java.io.Serializable;
+import java.util.UUID;
 
+@Embeddable
```

Comment:
How are this two entities connected, this one is embeddable and the other one has Embedded id from this one, how that works?

Reply:
UserObjectAccessEntity has a composite primary key

when your table has a composite primary key, you have use embeddedIds (though there are other ways to do this)
I've made UserObjectAccessKey an embeddable, and it serves as an embeddedId in UserObjectAccessEntity

Reply:
unfortunately, this composite key happens to be made up of two foreign keys
you can use "@MapsId" to map these foreign keys to field in "@EmbeddedId"

Reply:
we need this because the primary key is composite
```
@EmbeddedId private UserObjectAccessKey id;
```

we need this because these are foreign keys
```
@MapsId("grantedToId")
@ManyToOne
@JoinColumn(name = "granted_to", nullable = false)
private UserEntity grantedTo;

@MapsId("objectId")
@ManyToOne
@JoinColumn(name = "object_id", nullable = false)
private ObjectEntity object;
```